browser-extension/mv3/src/page-scripts/ajaxRequestInterceptor/fetch.js

```diff
@@ -179,7 +181,11 @@ export const initFetchInterceptor = (debug) => {
           };
         }
 
-        customResponse = getFunctionFromCode(responseModification.value, "response")(evaluatorArgs);
+        const isUserFunctionValid = getFunctionFromCode(responseModification.value, "response");
+
+        if (isUserFunctionValid) {
+          customResponse = generateUserFunctionWithSharedState(responseModification.value, debug)(evaluatorArgs);
+        }
```

Comment:
Why can't we just use `getFunctionFromCode` (which implicitly contains shared state in the closure). Why do we need to check whether function is valid first?

Reply:
Kept it separate that so that `notifyOnErrorOccurred` is not called each time there's `generateUserFunctionWithSharedState`. There could be instances when we do not want to call `notifyOnErrorOccurred` intentionally